### PR TITLE
TN-398 Notification & UserNotification models, ToU/Privacy Notifications for ePROMs

### PR DESCRIPTION
**Pull request by OptimusRhine**
* created Notification class
  * stores unique name field and content field (which is what will be displayed to users)
  * includes from and as JSON methods
* created UserNotification class
  * connects Notifications to Users via `User.notifications` relationship
* added relevant migrations
* created API endpoints for UserNotifications
  * `/api/user/<user_id>/notification [GET]` retrieves all the user's `User.notifications` in JSON format
  * `/api/user/<user_id>/notification/<notification_id> [DELETE]` removes the UserNotification object for the given user_id and notification_id (note the actual Notification object remains)
* added unit tests for Notification models and API endpoints
* added Notifications to list of objects for SitePersistence import
* added Terms and PrivacyPolicy Update Notifications to ePROMs config JSON
* created new task to generate UserNotifications for configured Notification and Roles list (i.e. all non-deleted users with any of the provided Roles will have the given Notification added to their `User.notifications` list)
* added Terms and PrivacyPolicy Update Notify ScheduledJobs to ePROMs config JSON

----------

**NOTE:** Please do **NOT** merge until @achen2401 has completed the UI portion of this task. Just creating PR now so that my changes can be reviewed today.

### CODE REVIEW

**Comment by achen2401** [comment]
@OptimusRhine Adam, I have made my front end changes. I only QAed it myself.  It would be great if you can have a look and see if my changes work for you as well?  Thank you very much in advance.

**Comment by ivan-c** [comment]
@achen2401 Just merged your JS refactor; I'm still reviewing this, but could you fix the [conflicts here](https://github.com/uwcirg/true_nth_usa_portal/pull/1753/conflicts)?
Thanks!

**Comment by achen2401** [comment]
@ivan-c  Ivan, conflicts here should be resolved.  Let me know if you encounter issue or see any problem with my fixes.

File: portal/static/js/main.js [hunk 1]
```diff
@@ -1280,6 +1296,76 @@ var fillContent = {
...
+            var notifications = [];
+            var notificationText = "";
+            (data.notifications).forEach(function(notice) {
+                notificationText += "<div class='notification' data-id='" + notice.id + "'>" + i18next.t(notice.content) + "</div>";
+            });
+            if (hasValue(notificationText)) {
+
+                var infoText = "<div class='notification-info'><i class='fa fa-info-circle' aria-hidden='true'></i>";
+
+                if (data.notifications.length > 1) {
```
Comment (ivan-c) [inline portal/static/js/main.js]: @achen2401 Just a heads up, we'll likely have to better implement [plural forms](https://www.i18next.com/plurals.html)

File: portal/views/notification.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        format: int64
+    produces:
+      - application/json
+    responses:
+      200:
+        description: successful operation
+        schema:
+          id: user_notifications
+          properties:
+            notifications:
```
Comment (ivan-c) [inline portal/views/notification.py]: I think we typically wrap multiple resources in a Bundle, are we still doing that? @pbugni

**Comment by ivan-c** [comment]
@achen2401 Sorry, I merged your other PR and introduced conflicts in this one, can you resolve [them](https://github.com/uwcirg/true_nth_usa_portal/pull/1753/conflicts)?

**Pull request merged**